packages/ui/components/dialogs/loop/index.tsx

```diff
@@ -374,10 +505,93 @@ export default function Loop({
     ]);
 
     try {
-      const tx = await currentSdk?.adjustLeverageRatio(
-        currentPosition?.address ?? ('' as Address),
-        currentLeverage
-      );
+      const previewDeposit = await publicClient.readContract({
+        abi: icErc20Abi,
+        address: selectedCollateralAsset.cToken,
+        functionName: 'previewDeposit',
+        args: [amountAsBInt]
+      });
+
+      const actualRedeemedAmountForAggregatorSwap =
+        await publicClient.readContract({
+          abi: icErc20Abi,
+          address: selectedCollateralAsset.cToken,
+          functionName: 'previewRedeem',
+          args: [previewDeposit]
+        });
+
+      // get initial quote to calculate slippage
+      const [, initialBorrowAmount] = await publicClient.readContract({
+        abi: iLeveredPositionFactoryAbi,
+        address: factory.address,
+        functionName: 'calculateAdjustmentAmountDeltas',
+        args: [
+          true,
+          parseEther(currentLeverage.toString()),
+          selectedCollateralAsset.underlyingPrice,
+          selectedBorrowAsset!.underlyingPrice,
+          1n,
+          actualRedeemedAmountForAggregatorSwap,
+          0n
+        ]
+      });
+
+      const quote = await getQuote({
+        fromChain: chainId,
+        toChain: chainId,
+        fromToken: selectedBorrowAsset!.underlyingToken,
+        toToken: selectedCollateralAsset.underlyingToken,
+        fromAmount: initialBorrowAmount.toString(),
```

Comment:
when the ratio is decreased, the swap amount (`fromAmount` param) is derived from the first output variable `supplyAmount` of `calculateAdjustmentAmountDeltas`, because the redeemed collateral should be exchanged for the other asset in order to repay the flash loan